tests/test__mongomock.py

```diff
@@ -318,6 +319,14 @@ def test__default_fields_to_id_if_empty(self):
         self.cmp.do.insert({'name':'Chucky', 'type':'doll', 'model':'v6'})
         self.cmp.compare_ignore_order.find({'name':'Chucky'}, fields = [])
 
+    def test__do_not_return_objects_without_requested_field(self):
+        self.cmp.do.insert({'name':'Chucky', 'type':'doll', 'model':'v6'})
+	self.cmp.compare.find_one({'name':'Chucky'}, fields = {'surname': True, '_id': False})
```

Comment:
Whoops. It looks like I forgot about this and deleted my fork, so I don't have an easy way to update this pull request anymore. If someone else wants to recreate and submit my changes, they're more than welcome.